app/controllers/users_controller.rb

```diff
@@ -26,4 +37,13 @@ def show
   def user_params
     params.require(:user).permit!
   end
+
+  def handle_invitation
+    if params[:invitation_token]
+      invitation = Invitation.find_by(token: params[:invitation_token])
+      invitation.inviter.follow(@user)
+      @user.follow(invitation.inviter)
+      invitation.update_column(:token, nil)
```

Comment:
It's a little bit more tricky, you have to get rid of `before_create` callback in `Tokenable` and adjust it  the solution accordingly. When you mix-in `Tokenable` to `Invitation` model, i would recommend using `after_create` - check my reply on forums :)